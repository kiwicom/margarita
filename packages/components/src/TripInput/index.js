// @flow

export { default as TripInput } from './TripInput';
