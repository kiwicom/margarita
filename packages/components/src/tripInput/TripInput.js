// @flow strict

import * as React from 'react';
import { View, Platform } from 'react-native';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';
import {
  StyleSheet,
  Icon,
  designTokens,
  type StylePropType,
} from '@kiwicom/universal-components';

import TouchableWithoutFeedback from '../TouchableWithoutFeedback';
import Text from '../text/Text';

type Props = {|
  +style?: StylePropType,
  +onPress: () => void,
  +icon: React.Element<typeof Icon>,
  +label: string,
  +value: string,
  +placeholder?: string,
|};

export default function TripInput({
  style,
  icon,
  label,
  value,
  placeholder,
  onPress,
}: Props) {
  const inputIcon = React.cloneElement(icon, {
    color: defaultTokens.colorIconSecondary,
  });
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={[styles.container, style]}>
        {Platform.OS !== 'web' && <View style={styles.icon}>{inputIcon}</View>}
        <Text style={styles.label}>{label.length > 0 ? `${label}: ` : ''}</Text>
        {value !== '' && <Text style={styles.value}>{value}</Text>}
        {value === '' && placeholder != null && <Text>{placeholder}</Text>}
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginBottom: parseInt(defaultTokens.spaceXSmall, 10),
    padding: 11,
    alignItems: 'center',
    borderRadius: parseInt(defaultTokens.borderRadiusLarge, 10),
    backgroundColor: defaultTokens.backgroundButtonSecondary,
    web: {
      flex: 1,
      height: parseInt(defaultTokens.heightInputNormal, 10),
      borderRadius: 3,
      backgroundColor: defaultTokens.paletteWhite,
      borderWidth: 1,
      borderColor: defaultTokens.paletteInkLighter,
    },
  },
  icon: {
    marginEnd: 10,
  },
  label: {
    color: defaultTokens.paletteInkNormal,
    web: {
      color: defaultTokens.colorTextSecondary,
    },
  },
  value: {
    color: defaultTokens.colorTextAttention,
    web: {
      color: defaultTokens.paletteInkNormal,
      fontWeight: designTokens.fontWeightMedium,
    },
  },
});
