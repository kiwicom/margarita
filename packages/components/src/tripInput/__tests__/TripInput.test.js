// @flow

import * as React from 'react';
import { shallow } from 'react-native-testing-library';

import TripInput from '../TripInput';

const IconMock = () => null;

it('renders', () => {
  expect(
    shallow(
      <TripInput
        onPress={jest.fn()}
        icon={<IconMock />}
        label="label"
        value="value"
      />,
    ),
  ).toMatchSnapshot();
});

it('renders with placeholder', () => {
  expect(
    shallow(
      <TripInput
        onPress={jest.fn()}
        icon={<IconMock />}
        label="label"
        value=""
        placeholder="placeholder"
      />,
    ),
  ).toMatchSnapshot();
});

it('renders correctly with placeholder and value', () => {
  expect(
    shallow(
      <TripInput
        onPress={jest.fn()}
        icon={<IconMock />}
        label="label"
        value="value"
        placeholder="placeholder"
      />,
    ),
  ).toMatchSnapshot();
});
