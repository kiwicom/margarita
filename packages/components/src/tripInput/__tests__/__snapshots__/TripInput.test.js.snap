// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "output": <TouchableWithoutFeedback
    accessibilityComponentType="button"
    accessibilityTraits="button"
    onPress={[MockFunction]}
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#e8edf1",
            "borderRadius": 6,
            "flexDirection": "row",
            "marginBottom": 8,
            "padding": 11,
          },
          undefined,
        ]
      }
    >
      <Component
        style={
          Object {
            "marginEnd": 10,
          }
        }
      >
        <IconMock
          color="#7f91a8"
        />
      </Component>
      <Text
        style={
          Object {
            "color": "#46515e",
          }
        }
      >
        label: 
      </Text>
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Object {
            "color": "#171b1e",
          }
        }
      >
        value
      </Text>
    </Component>
  </TouchableWithoutFeedback>,
}
`;

exports[`renders correctly with placeholder and value 1`] = `
Object {
  "output": <TouchableWithoutFeedback
    accessibilityComponentType="button"
    accessibilityTraits="button"
    onPress={[MockFunction]}
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#e8edf1",
            "borderRadius": 6,
            "flexDirection": "row",
            "marginBottom": 8,
            "padding": 11,
          },
          undefined,
        ]
      }
    >
      <Component
        style={
          Object {
            "marginEnd": 10,
          }
        }
      >
        <IconMock
          color="#7f91a8"
        />
      </Component>
      <Text
        style={
          Object {
            "color": "#46515e",
          }
        }
      >
        label: 
      </Text>
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Object {
            "color": "#171b1e",
          }
        }
      >
        value
      </Text>
    </Component>
  </TouchableWithoutFeedback>,
}
`;

exports[`renders with placeholder 1`] = `
Object {
  "output": <TouchableWithoutFeedback
    accessibilityComponentType="button"
    accessibilityTraits="button"
    onPress={[MockFunction]}
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#e8edf1",
            "borderRadius": 6,
            "flexDirection": "row",
            "marginBottom": 8,
            "padding": 11,
          },
          undefined,
        ]
      }
    >
      <Component
        style={
          Object {
            "marginEnd": 10,
          }
        }
      >
        <IconMock
          color="#7f91a8"
        />
      </Component>
      <Text
        style={
          Object {
            "color": "#46515e",
          }
        }
      >
        label: 
      </Text>
      <Text>
        placeholder
      </Text>
    </Component>
  </TouchableWithoutFeedback>,
}
`;
